concourse/steps/version.mako

```diff
@@ -21,13 +22,22 @@ path_to_repo_version_file = os.path.join(
 output_version_file = os.path.join(job_step.output('version_path'), 'version')
 legacy_version_file = os.path.join(job_step.output('version_path'), 'number')
 
-if (read_callback := version_trait.read_callback()):
+# Assign empty string to calbacks if None, as we'd template 'None' (the string) later otherwise
+if (read_callback := version_trait.read_callback() or ''):
   read_callback = os.path.join(main_repo.resource_name(), read_callback)
 
-# Assign empty string if None, as we'd template 'None' (the string) later otherwise
+
 if (write_callback := version_trait.write_callback() or ''):
   write_callback = os.path.join(main_repo.resource_name(), write_callback)
 
+# Validate that either both callbacks are set or none
+if (write_callback == '' and read_callback != '') or (write_callback != '' and read_callback == ''):
+  raise ValueError(f"write_callback is '{write_callback}' and read_callback is '{read_callback}'. Either set both callbacks or none!")
+
+# Set version_interface to 'callback' if write_callback and read_callback are set
+if (write_callback != '' and read_callback != ''):
+  version_trait.raw['version_interface']='callback'
```

Comment:
similar to validation, I think it would be nicer to move this code into `concourse.model.traits.version`. More specifically into the "TraitProcessor" (an awkwardly-named hook that is called after parsing pipeline-definitions, but before executing the pipeline-rendering:

https://github.com/gardener/cc-utils/blob/master/concourse/model/traits/version.py#L127

I think processing might be done in an initialiser (would need to be defined). [example](https://github.com/gardener/cc-utils/blob/master/concourse/model/traits/component_descriptor.py#L457-L459). It might also be done earlier (e.g. in the trait's initialiser)

Reply:
also (nit): pep8 suggests there should be whitespace (blank) around assignment operator in this case (`] = 'call..`)

Reply:
done.